config/http_config.go

```diff
@@ -162,6 +165,33 @@ func NewRoundTripperFromConfig(cfg HTTPClientConfig, name string) (http.RoundTri
 	return rt, nil
 }
 
+type tlsConfigRoundTripper struct {
+	cfg       HTTPClientConfig
+	tlsConfig *tls.Config
+	name      string
+	rt        http.RoundTripper
+}
+
+// NewTLSConfigRoundTripper reads the tls configuration upton each request
+// and updates the TLS configuration as needed
+func NewTLSConfigRoundTripper(cfg HTTPClientConfig, tlsConfig *tls.Config, name string, rt http.RoundTripper) http.RoundTripper {
+	return &tlsConfigRoundTripper{cfg, tlsConfig, name, rt}
+}
+
+func (rt *tlsConfigRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	tlsConfig, err := NewTLSConfig(&rt.cfg.TLSConfig)
+	if err != nil {
+		return nil, err
+	}
+	if !reflect.DeepEqual(tlsConfig, rt.tlsConfig) {
```

Comment:
I think there's mutable state in a tls.Config, so this may always return true and cause spurious reconnects